modules/redisearch.ts

```diff
@@ -80,7 +80,7 @@ export class Redisearch extends Module {
      * @param parameters The additional optional parameter
      * @returns Array reply, where the first element is the total number of results, and then pairs of document id, and a nested array of field/value.
      */
-    async search(index: string, query: string, parameters?: FTSearchParameters): Promise<number> {
+    async search(index: string, query: string, parameters?: FTSearchParameters): Promise<[number, ...any]> {
```

Comment:
I am not sure this is correct, the handleResponse function kind of removes the brackets. Please print the results in the unit tests of redisearch and verify if all of the res types are correct.
In addition, we try our best in this project to not use 'any' type if possible. So it's better we minimized it to specific types if possible.

Reply:
I tried to fix the issue you mentioned. I hope it works 🤞🏻